packages/features/ee/integration-attribute-sync/services/IntegrationAttributeSyncService.ts

```diff
@@ -0,0 +1,90 @@
+import type { CredentialRepository } from "@calcom/features/credentials/repositories/CredentialRepository";
+
+import type { ZCreateAttributeSyncSchema } from "@calcom/trpc/server/routers/viewer/attribute-sync/createAttributeSync.schema";
+import { enabledAppSlugs } from "../constants";
+import type { IIntegrationAttributeSyncRepository } from "../repositories/IIntegrationAttributeSyncRepository";
+import { type ISyncFormData, attributeSyncRuleSchema } from "../schemas/zod";
+
+interface IIntegrationAttributeSyncServiceDeps {
+  credentialRepository: CredentialRepository;
+  integrationAttributeSyncRepository: IIntegrationAttributeSyncRepository;
+}
+
+export class IntegrationAttributeSyncService {
+  constructor(private readonly deps: IIntegrationAttributeSyncServiceDeps) {}
+
+  async getEnabledAppCredentials(organizationId: number) {
+    return this.deps.credentialRepository.findByTeamIdAndSlugs({
+      teamId: organizationId,
+      slugs: enabledAppSlugs,
+    });
+  }
+
+  async getAllIntegrationAttributeSyncs(organizationId: number) {
+    return this.deps.integrationAttributeSyncRepository.getByOrganizationId(organizationId);
+  }
+
+  async getById(id: string) {
+    return this.deps.integrationAttributeSyncRepository.getById(id);
+  }
+
+  async createAttributeSync(input: ZCreateAttributeSyncSchema, organizationId: number) {
+    const credential = await this.deps.credentialRepository.findByIdAndTeamId({
+      id: input.credentialId,
+      teamId: organizationId,
+    });
+
+    if (!credential) {
+      throw new Error("Credential not found");
+    }
+
+    const parsedRule = attributeSyncRuleSchema.parse(input.rule);
+
+    return this.deps.integrationAttributeSyncRepository.create({
+      name: input.name,
+      organizationId,
+      integration: credential.app?.slug || credential.type,
+      credentialId: input.credentialId,
+      enabled: input.enabled,
+      rule: parsedRule,
+      syncFieldMappings: input.syncFieldMappings,
+    });
+  }
+
+  async updateIncludeRulesAndMappings(data: ISyncFormData) {
+    const { syncFieldMappings, rule, ruleId, ...integrationAttributeSync } = data;
+    const existingFieldMappings = await this.deps.integrationAttributeSyncRepository.getSyncFieldMappings(
+      data.id
+    );
+
+    const parsedRule = attributeSyncRuleSchema.parse(rule);
+
+    const incomingMappingIds = new Set(
+      syncFieldMappings.reduce((ids, mapping) => {
+        if ("id" in mapping) ids.push(mapping.id);
+        return ids;
+      }, [] as string[])
+    );
```

Comment:
The data consists of field mappings with an id (already exists and needs to be updated) or without an id (needs to be created)